modules/core/src/main/scala/org/scalasteward/core/nurture/NurtureAlg.scala

```diff
@@ -74,12 +84,24 @@ final class NurtureAlg[F[_]](
       nonSbtUpdates <- getNonSbtUpdates(repo)
       updates = sbtUpdates ::: nonSbtUpdates
       filtered <- filterAlg.localFilterMany(repoConfig, updates)
+      dependencies <- sbtAlg.getDependencies(repo)
+      filteredDependencies = dependenciesInUpdates(dependencies, filtered)
+      mapping <- coursierAlg.getArtifactIdUrlMapping(filteredDependencies)
```

Comment:
In https://github.com/fthomas/scala-steward/pull/762#issuecomment-520037653 I meant to call  `coursierAlg.getArtifactIdUrlMapping` only in `NurtureAlg.createPullRequest` because only there these URLs are needed. There is no need to call Coursier if a pull request already exists and Scala Steward only updates/ignores it.

Reply:
I see.
I think it is addressed in https://github.com/fthomas/scala-steward/pull/762/commits/2b3d60f45ed5f07930f1ac71984c2714ce907fcc

Reply:
Thanks! What I didn't realized until now is that the `sbtAlg.getDependencies` call is needed to get artifactIds with the cross version suffix and that we don't get them from `Update` because that only contains artifactIds without that suffix. Unfortunately `sbtAlg.getDependencies` is a expensive call because it runs sbt. Calling it in `createPullRequest` now means that we potentially call `getDependencies` more than once for the same project although it would always return the same result.

There are a few options to handle this:
* Find a way to add artifactIds with cross version suffixes to `Update`. I've no idea if we can get this information from sbt-updates.
* Use `RepoCacheAlg` because dependencies are already available in `RepoCache`. Unfortunately this is only true if `Config.pruneRepos` is `true`. If we want to use `RepoCacheAlg` in `NurtureAlg`, we would have to ensure that the caches are fresh even if `Config.pruneRepos` is false.
* Memoize the result of  `sbtAlg.getDependencies` so that it is called at most once.
* Go back to 2b3d60f45ed5f07930f1ac71984c2714ce907fcc so that `sbtAlg.getDependencies` is called once for every repo.
* Leave it as is and handle this in another PR to prevent more back and forth in this PR.

I really would prefer the first option but don't know if it is feasible at all. It would also require its own PR. To not further delay this feature I'm also okay with the last option. What do you think, @exoego?

And sorry for not realizing this earlier!

Reply:
I think 3rd option (memoize) is good for now since it is not difficult to implement and do not degrade runtime speed, and runs on any configuration.
Then let's find a way to realize 1st option. (maybe open PR to sbt-update?)

I will implement memoize in SbtAlg using `mutable.Map.getOrElseUpdate`, but suggestion is appreciated if you have a better design.
Ofcourse it is OK to merge as-is.

Reply:
You could use `cats.effect.Async.memoize` to memoize `getDependencies` in `NurtureAlg`. I think there is no need to modify `SbtAlg` for that.

For the 1st option I'm currently looking at sbt-updates' `dependencyUpdatesData` task which returns a `Map[ModuleID, SortedSet[Version]]`. Maybe the `ModuleID` there is enough to get the artifactIds with the cross version suffix.

Reply:
Added memoize in https://github.com/fthomas/scala-steward/pull/762/commits/80ee906d3b00575ff7e814dfc142eee249db0a55